### PR TITLE
fix: group-avatar component usage

### LINKED ISSUE #17013: update group-avatar component
# Problem

Follow-up of #16931 
Update the `size` property to receive any valid number

## Implementation

Refactor group-avatar component to receive any valid number as the `size` property

### PR DESCRIPTION
**Pull request by codemaster115**
fixes #17013 

### Summary

Update usage of `group-avatar` component

status: ready

### CODE REVIEW

**Review by J-Son89 — APPROVED** [review]

**Comment by J-Son89** [comment]
@qoqobolo @smohamedjavid @0xngmare - not sure if you have been discussing with each other. 

Before checking https://github.com/status-im/status-mobile/pull/17039
Perhaps it's easier to verify this pr fixes the issues addressed as we will have to make these changes anyway 👍

File: src/status_im2/common/confirmation_drawer/view.cljs [hunk 1]
```diff
@@ -11,7 +11,7 @@
   (if group-chat
     [quo/group-avatar
      {:customization-color color
-      :size                :x-small}]
+      :size                20}]
```
Comment (smohamedjavid) [inline src/status_im2/common/confirmation_drawer/view.cljs]: `x-small` and `small` using `20`. Is it expected?

**Review by smohamedjavid — COMMENTED** [review]
Apologies, Can we wait until the PR https://github.com/status-im/status-mobile/pull/17039 reverts the changes? 

This will help us review your PR as in whole (size API updates and usages) and resolve/debug any issues if encountered.

File: src/status_im2/common/confirmation_drawer/view.cljs (hunk 1 above)
Comment (codemaster115) [reply src/status_im2/common/confirmation_drawer/view.cljs]: That's not expected, but current components are all using 20 as the avatar size.

**Comment by qoqobolo** [comment]
Hey @J-Son89 @smohamedjavid @codemaster115, so I'm checking https://github.com/status-im/status-mobile/pull/17039 first, and then we'll test this one 👍

**Review by smohamedjavid — APPROVED** [review]

File: src/quo2/components/avatars/group_avatar/view.cljs [hunk 2]
```diff
@@ -7,34 +7,27 @@
...
-               :small   16
-               :medium  16
-               :large   20
-               :x-large 32}
-   :container {:x-small 20
-               :small   28
-               :medium  32
-               :large   48
-               :x-large 80}})
+  {20 {:icon 12}
```
Comment (J-Son89) [inline src/quo2/components/avatars/group_avatar/view.cljs]: I think we should wait for the outcome of the current discussion before merging this pr as we would need to update it's uses here anyway 👍

**Comment by qoqobolo** [comment]
> I think we should wait for the outcome of the current discussion before merging this pr as we would need to update it's uses here anyway 👍

@J-Son89 @codemaster115 Please, ping the QA team once it's ready to be tested.

**Comment by codemaster115** [comment]
@qoqobolo This PR is ready to test.

File: src/quo2/components/avatars/group_avatar/view.cljs [hunk 3]
```diff
@@ -7,23 +7,23 @@
...
+  {:icon      {:size/xs-20 12
+               :size/sm-28 16
+               :size/md-32 16
+               :size/lg-48 20
+               :size/xl-80 32}
+   :container {:size/xs-20 20
+               :size/sm-28 28
+               :size/md-32 32
+               :size/lg-48 48
+               :size/xl-80 80}})
```
Comment (ajayesivan) [inline src/quo2/components/avatars/group_avatar/view.cljs]: @codemaster115 I'm a bit confused here, `:size/xs-20` is `12` for icons & `20` for containers, then why we are using `-20` in both keywords?

cc: @J-Son89

File: src/quo2/components/avatars/group_avatar/view.cljs [hunk 4]
```diff
@@ -7,23 +7,23 @@
...
 
 (defn- view-internal
   [_]
   (fn [{:keys [size theme customization-color picture]
-        :or   {size                :x-small
-               customization-color :blue
-               picture             nil}}]
+        :or   {size                :size/xs-20
+               customization-color :blue}}]
+    (println size)
```
Comment (ajayesivan) [inline src/quo2/components/avatars/group_avatar/view.cljs]: Please remove this print statement.

File: src/quo2/components/avatars/group_avatar/view.cljs (hunk 3 above)
Comment (J-Son89) [reply src/quo2/components/avatars/group_avatar/view.cljs]: -20 is referring to the size property marked in figma. Lots of components have this property to handle the different variants.

**Comment by qoqobolo** [comment]
Hey @codemaster115, thanks for your work!
Could you pls confirm that this is the correct group avatar size?
It looks too small, but probably that's expected for now. 

Screenshot 2023-08-21 at 10 42 48
Screenshot 2023-08-21 at 10 43 01

File: src/quo2/components/avatars/group_avatar/view.cljs (hunk 3 above)
Comment (smohamedjavid) [reply src/quo2/components/avatars/group_avatar/view.cljs]: > @codemaster115 I'm a bit confused here, :size/xs-20 is 12 for icons & 20 for containers, then why we are using -20 in both keywords?

Me too @ajayesivan.

@codemaster115 - We can merge the size map to be more cleaner.

```clojure
{:size/xs-20 {:container-size 20
              :icon-size      12}
{:size/sm-28 {:container-size 28
              :icon-size      16}
```

<br />
<hr />

Not a fan of using `xs`, `sm`, `lg`, `xl`, ...etc with size numbers.

From my knowledge, the lowest size for a component is `14` in the `preview-list` component. So, I would presumably assign `14` for `xxxs-14`. But, If there is any other component that has size `12`, the `xxxs` would mean that. 

Can we use a common key like `:size/size-20` or `:size/s-20`?

Comment (J-Son89) [reply src/quo2/components/avatars/group_avatar/view.cljs]: Yep I agree to both points here @smohamedjavid!

File: src/quo2/components/avatars/group_avatar/view.cljs [hunk 5]
```diff
@@ -7,25 +7,24 @@
...
+  {:size/s-20 {:icon      12
+               :container 20}
+   :size/s-28 {:icon      16
+               :container 28}
+   :size/s-32 {:icon      16
+               :container 32}
+   :size/s-48 {:icon      20
+               :container 48}
+   :size/s-80 {:icon      32
+               :container 80}})
```
Comment (ajayesivan) [inline src/quo2/components/avatars/group_avatar/view.cljs:19]: ❤️

**Review by ajayesivan — APPROVED** [review]
LGTM! Nice work @codemaster115

**Comment by codemaster115** [comment]
> Hey @codemaster115, thanks for your work! Could you pls confirm that this is the correct group avatar size? It looks too small, but probably that's expected for now.
> 
> Screenshot 2023-08-21 at 10 42 48 Screenshot 2023-08-21 at 10 43 01

@qoqobolo Please test it again

**Review by smohamedjavid — APPROVED** [review]
Nice work! 🎉

**Comment by qoqobolo** [comment]
Thanks for your work @codemaster115!
PR is ready for design review @Francesca-G

**Review by Francesca-G — APPROVED** [review]

**Pull request merged**